src/components/themed/FilledTextInput.tsx

```diff
@@ -197,7 +198,7 @@ export const FilledTextInput = React.forwardRef<FilledTextInputRef, FilledTextIn
   function clear(): void {
     if (inputRef.current != null) inputRef.current.clear()
     if (onChangeText != null) onChangeText('')
-    if (blurOnClear || !hasValue) blur()
+    if (blurOnClear || valueRef.current.length === 0) blur()
```

Comment:
Optional: `value.current === ''` would be simpler. You use the `value.current.length > 0` pattern in many places, and I feel like it's harder to read - I have to stop & think about math for a split second, vs. it literally just being `''` or not.